### PR TITLE
Enhance debug, trace & info logs for remote store flows

### PR DESCRIPTION
**Pull request by ashking94**
### Description
This PR improves the logging by adding debug, info and trace logs. 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: libs/core/src/main/java/org/opensearch/core/index/Index.java [hunk 1]
```diff
@@ -59,6 +59,7 @@ public class Index implements Writeable, ToXContentObject {
     public static final Index[] EMPTY_ARRAY = new Index[0];
     private static final String INDEX_UUID_KEY = "index_uuid";
     private static final String INDEX_NAME_KEY = "index_name";
+    public static final String UNKNOWN_INDEX_NAME = "unknown";
```
Comment (sachinpkale) [inline libs/core/src/main/java/org/opensearch/core/index/Index.java:62]: What happens if an index is created with name `unknown`?

File: server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java [hunk 2]
```diff
@@ -294,6 +294,11 @@ public void updateLatestLocalFileNameLengthMap(
         Collection<String> segmentFiles,
         CheckedFunction<String, Long, IOException> fileSizeFunction
     ) {
+        logger.debug(
```
Comment (sachinpkale) [inline server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java:297]: In OpenSearch codebase, in bunch of places, we have a check before logging with debug or trace level.

```
if (logger.isDebugEnabled()) {
    logger.debug("Some text");
}

```

I am not completely sure how it helps but better to check further.

File: server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java [hunk 3]
```diff
@@ -111,13 +116,15 @@ public RemoteSegmentStoreDirectory(
...
-        ThreadPool threadPool
+        ThreadPool threadPool,
+        ShardId shardId
     ) throws IOException {
         super(remoteDataDirectory);
         this.remoteDataDirectory = remoteDataDirectory;
         this.remoteMetadataDirectory = remoteMetadataDirectory;
         this.mdLockManager = mdLockManager;
         this.threadPool = threadPool;
+        this.logger = Loggers.getLogger(getClass(), shardId);
```
Comment (sachinpkale) [inline server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java:127]: Nice! This was much needed.

File: server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java [hunk 4]
```diff
@@ -262,6 +262,7 @@ public void onFailure(Exception e) {
         updateFinalStatusInSegmentTracker(successful.get(), bytesBeforeUpload, startTimeInNS);
         // If there are failures in uploading segments, then we should retry as search idle can lead to
         // refresh not occurring until write happens.
+        logger.debug("syncSegments runStatus={}", successful.get());
```
Comment (sachinpkale) [inline server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java:265]: In `onSuccessfulSegmentsSync`, we set the min seq number to keep. If it is not logged in translog, we may need to log that as well.

File: server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java (hunk 2 above)
Comment (sachinpkale) [reply server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java:297]: log4j doc mentions that this is not required anymore: https://github.com/apache/logging-log4j2/blob/2.x/log4j-api/src/main/java/org/apache/logging/log4j/Logger.java#L56

Comment (ashking94) [reply server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java:297]: Thanks for checking @sachinpkale.

File: libs/core/src/main/java/org/opensearch/core/index/Index.java (hunk 1 above)
Comment (ashking94) [reply libs/core/src/main/java/org/opensearch/core/index/Index.java:62]: Possible, let me rename it to `_unknown_` as we don't allow it to start with underscore.

File: server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java (hunk 4 above)
Comment (ashking94) [reply server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java:265]: Done

**Comment by ashking94** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/26311/
> * **CommitID:** [d7ff04d](https://github.com/opensearch-project/OpenSearch/commit/d7ff04ddacfbbbafc3196358a190c48fa3e95f9f)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Flaky test - https://github.com/opensearch-project/OpenSearch/issues/10154

**Review by sachinpkale — APPROVED** [review]

**Comment by ashking94** [comment]
Auto backport failed. Raised manual backport - https://github.com/opensearch-project/OpenSearch/pull/10237. Conflicts came up due to https://github.com/opensearch-project/OpenSearch/pull/9710 not being backported.

**Pull request merged**